src/core/mesh/qgsmeshlayer.cpp

```diff
@@ -73,6 +73,22 @@ void QgsMeshLayer::setDefaultRendererSettings()
   }
 }
 
+void QgsMeshLayer::createSimplifiedMeshes()
+{
+  if ( mSimplifySettings.isEnabled() && mTriangularMeshes.size() == 1 )
+  {
+    double reductionFactor = mSimplifySettings.reductionFactor();
+
+    QVector<QgsTriangularMesh *> simplifyMeshes =
+      mTriangularMeshes[0]->simplifyMesh( reductionFactor );
```

Comment:
I am starting to question if the base triangular mesh shouldn't have its own member variable and not be in vector with simplified meshes.. the code would be cleaner

Reply:
I agree the readability will be greater, but the code will be a bit more complicated. Because the base mesh is a mesh like simplified meshes, especially when rendering. So I am not sure it will be more cleaner, but I am open for discussion...

Reply:
ok